src/adapter/sources.ts

```diff
@@ -852,23 +852,31 @@ export class SourceContainer {
     contentHash?: string,
   ): Promise<Source> {
     const absolutePath = await this.sourcePathResolver.urlToAbsolutePath({ url });
+    let absoluteSourceMapPath = undefined;
+    if (
+      sourceMapUrl &&
+      this.sourcePathResolver.shouldResolveSourceMap({
+        sourceMapUrl,
+        compiledPath: absolutePath || url,
+      })
+    ) {
+      absoluteSourceMapPath = await this.sourcePathResolver.urlToAbsolutePath({
```

Comment:
The source map might not always exist at an absolute path on disk. For example, when using the webpack-dev-server. That would cause this to be undefined, or an invalid path.

Did the initial code pointer to making the change within the `SourceMapFactory` not work out? If we make the change in there, then we could also easily add fallback logic to avoid breaking existing configurations.

Reply:
Used this way to keep this as much as possible like the pathMapping of the original file source. But I can give it another shot with the code points you provided tomorrow.

Reply:
I pushed another commit. `SourceMapFactory` now tries to load from the path mapped path first, but if that's unsuccessful, loading from the original path is attempted. Also logs a trace message if this happens.